From 144b06c95dd038138d5b05f23611e5af9d504efd Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Thu, 26 Apr 2018 14:39:39 +0100
Subject: [PATCH 3/7] lineage-sdk lights: Genericize adjustable brightness
 capability

 * Since every lights HAL supporting brightness adjustment should
   properly handle both battery and notification lights, rename
   this capability to make it generic in order to avoid confusion.

Change-Id: I03bc0c60f964e9eb47034712ab4229c240266d65
---
 lineage/res/res/values/config.xml                         | 2 +-
 .../internal/notification/LightsCapabilities.java         | 6 +++---
 .../internal/notification/LineageBatteryLights.java       | 5 +++--
 .../internal/notification/LineageNotificationLights.java  | 8 +++++---
 4 files changed, 12 insertions(+), 9 deletions(-)

diff --git a/lineage/res/res/values/config.xml b/lineage/res/res/values/config.xml
index ab424b7..4a39cfe 100644
--- a/lineage/res/res/values/config.xml
+++ b/lineage/res/res/values/config.xml
@@ -65,7 +65,7 @@
          LIGHTS_MULTIPLE_NOTIFICATION_LED = 4 (deprecated)
          LIGHTS_PULSATING_LED = 8
          LIGHTS_SEGMENTED_BATTERY_LED = 16
-         LIGHTS_ADJUSTABLE_NOTIFICATION_LED_BRIGHTNESS = 32
+         LIGHTS_ADJUSTABLE_LED_BRIGHTNESS = 32
          LIGHTS_BATTERY_LED = 64
 
          For example, a device with notification and battery lights that supports
diff --git a/sdk/src/java/org/lineageos/internal/notification/LightsCapabilities.java b/sdk/src/java/org/lineageos/internal/notification/LightsCapabilities.java
index a1ee26a..91aab3c 100644
--- a/sdk/src/java/org/lineageos/internal/notification/LightsCapabilities.java
+++ b/sdk/src/java/org/lineageos/internal/notification/LightsCapabilities.java
@@ -1,5 +1,5 @@
 /**
- * Copyright (C) 2017 The LineageOS Project
+ * Copyright (C) 2017-2018 The LineageOS Project
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -36,8 +36,8 @@ public final class LightsCapabilities {
     // segments to show (in order to represent battery level).
     public static final int LIGHTS_SEGMENTED_BATTERY_LED = 16;
 
-    // The notification light supports adjustable brightness.
-    public static final int LIGHTS_ADJUSTABLE_NOTIFICATION_LED_BRIGHTNESS = 32;
+    // Device's lights HAL handles brightness adjustment.
+    public static final int LIGHTS_ADJUSTABLE_LED_BRIGHTNESS = 32;
 
     // Device has a battery light.
     public static final int LIGHTS_BATTERY_LED = 64;
diff --git a/sdk/src/java/org/lineageos/internal/notification/LineageBatteryLights.java b/sdk/src/java/org/lineageos/internal/notification/LineageBatteryLights.java
index a318ce1..324311f 100644
--- a/sdk/src/java/org/lineageos/internal/notification/LineageBatteryLights.java
+++ b/sdk/src/java/org/lineageos/internal/notification/LineageBatteryLights.java
@@ -80,10 +80,11 @@ public final class LineageBatteryLights {
         mMultiColorBatteryLed = LightsCapabilities.supports(
                 mContext, LightsCapabilities.LIGHTS_RGB_BATTERY_LED);
 
+        // Does the lights HAL handle brightness adjustment?
         mHALAdjustableBrightness = LightsCapabilities.supports(
-                mContext, LightsCapabilities.LIGHTS_ADJUSTABLE_NOTIFICATION_LED_BRIGHTNESS);
+                mContext, LightsCapabilities.LIGHTS_ADJUSTABLE_LED_BRIGHTNESS);
 
-        // We support brightness adjustment if either the HAL supports it
+        // Brightness adjustment is supported if either the lights HAL handles it
         // or the light is RGB adjustable.
         mCanAdjustBrightness = mHALAdjustableBrightness || mMultiColorBatteryLed;
 
diff --git a/sdk/src/java/org/lineageos/internal/notification/LineageNotificationLights.java b/sdk/src/java/org/lineageos/internal/notification/LineageNotificationLights.java
index 7337754..40ffc34 100644
--- a/sdk/src/java/org/lineageos/internal/notification/LineageNotificationLights.java
+++ b/sdk/src/java/org/lineageos/internal/notification/LineageNotificationLights.java
@@ -1,6 +1,6 @@
 /**
  * Copyright (C) 2015 The CyanogenMod Project
- * Copyright (C) 2017 The LineageOS Project
+ * Copyright (C) 2017-2018 The LineageOS Project
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -95,8 +95,9 @@ public final class LineageNotificationLights {
 
         final Resources res = mContext.getResources();
 
+        // Does the lights HAL handle brightness adjustment?
         mHALAdjustableBrightness = LightsCapabilities.supports(
-                mContext, LightsCapabilities.LIGHTS_ADJUSTABLE_NOTIFICATION_LED_BRIGHTNESS);
+                mContext, LightsCapabilities.LIGHTS_ADJUSTABLE_LED_BRIGHTNESS);
 
         mDefaultNotificationColor = res.getColor(
                 com.android.internal.R.color.config_defaultNotificationColor);
@@ -105,10 +106,11 @@ public final class LineageNotificationLights {
         mDefaultNotificationLedOff = res.getInteger(
                 com.android.internal.R.integer.config_defaultNotificationLedOff);
 
+        // Does the device support changing notification LED colors?
         mMultiColorNotificationLed = LightsCapabilities.supports(
                 mContext, LightsCapabilities.LIGHTS_RGB_NOTIFICATION_LED);
 
-        // We support brightness adjustment if either the HAL supports it
+        // Brightness adjustment is supported if either the lights HAL handles it
         // or the light is RGB adjustable.
         mCanAdjustBrightness = mHALAdjustableBrightness || mMultiColorNotificationLed;
 
-- 
2.17.0

